passphrase/passphrase.go

```diff
@@ -134,8 +123,19 @@ func (br *boundRetriever) requestPassphrase(keyName, alias string, createNew boo
 		fmt.Fprintf(br.out, "Enter passphrase for %s key%s: ", displayAlias, withID)
 	}
 
+	var (
+		passphrase []byte
+		err        error
+	)
+
 	stdin := bufio.NewReader(br.in)
-	passphrase, err := stdin.ReadBytes('\n')
+
+	if terminal.IsTerminal(int(os.Stdin.Fd())) {
```

Comment:
thanks for adding this!  Since we repeat this same code a few times, could we factor it out into a helper function?